src/code/z_kankyo.c

```diff
@@ -1941,107 +2050,132 @@ void Environment_PlaySceneSequence(GlobalContext* globalCtx) {
+
+            if (globalCtx->envCtx.precipitation[PRECIP_RAIN_MAX] == 0 &&
+                globalCtx->envCtx.precipitation[PRECIP_SOS_MAX] == 0) {
                 osSyncPrintf("\n\n\nNa_StartMorinigBgm\n\n");
                 func_800F5510(globalCtx->sequenceCtx.seqId);
             }
-            globalCtx->envCtx.unk_E0++;
+
+            globalCtx->envCtx.timeSeqState++;
             break;
-        case 1:
-            if (gSaveContext.dayTime > 0xB71C) {
-                if (globalCtx->envCtx.unk_EE[0] == 0 && globalCtx->envCtx.unk_F2[0] == 0) {
+
+        case TIMESEQ_FADE_DAY_BGM:
+            if (gSaveContext.dayTime > CLOCK_TIME(17, 10)) {
+                if (globalCtx->envCtx.precipitation[PRECIP_RAIN_MAX] == 0 &&
+                    globalCtx->envCtx.precipitation[PRECIP_SOS_MAX] == 0) {
                     Audio_QueueSeqCmd(0x1 << 28 | SEQ_PLAYER_BGM_MAIN << 24 | 0xF000FF);
                 }
-                globalCtx->envCtx.unk_E0++;
+
+                globalCtx->envCtx.timeSeqState++;
             }
             break;
-        case 2:
-            if (gSaveContext.dayTime > 0xC000) {
+
+        case TIMESEQ_NIGHT_BEGIN_SFX:
+            if (gSaveContext.dayTime > CLOCK_TIME(18, 0)) {
                 func_800788CC(NA_SE_EV_DOG_CRY_EVENING);
-                globalCtx->envCtx.unk_E0++;
+                globalCtx->envCtx.timeSeqState++;
             }
             break;
-        case 3:
-            if (globalCtx->envCtx.unk_EE[0] == 0 && globalCtx->envCtx.unk_F2[0] == 0) {
+
+        case TIMESEQ_EARLY_NIGHT_CRITTERS:
+            if (globalCtx->envCtx.precipitation[PRECIP_RAIN_MAX] == 0 &&
+                globalCtx->envCtx.precipitation[PRECIP_SOS_MAX] == 0) {
                 Audio_PlayNatureAmbienceSequence(globalCtx->sequenceCtx.natureAmbienceId);
                 Audio_SetNatureAmbienceChannelIO(NATURE_CHANNEL_CRITTER_0, CHANNEL_IO_PORT_1, 1);
             }
-            globalCtx->envCtx.unk_E0++;
+
+            globalCtx->envCtx.timeSeqState++;
             break;
-        case 4:
-            if (gSaveContext.dayTime > 0xCAAB) {
-                globalCtx->envCtx.unk_E0++;
+
+        case TIMESEQ_NIGHT_DELAY:
+            if (gSaveContext.dayTime > CLOCK_TIME(19, 0) + 1) {
+                globalCtx->envCtx.timeSeqState++;
             }
             break;
-        case 5:
+
+        case TIMESEQ_NIGHT_CRITTERS:
             Audio_SetNatureAmbienceChannelIO(NATURE_CHANNEL_CRITTER_0, CHANNEL_IO_PORT_1, 0);
-            if (globalCtx->envCtx.unk_EE[0] == 0 && globalCtx->envCtx.unk_F2[0] == 0) {
+
+            if (globalCtx->envCtx.precipitation[PRECIP_RAIN_MAX] == 0 &&
+                globalCtx->envCtx.precipitation[PRECIP_SOS_MAX] == 0) {
                 Audio_SetNatureAmbienceChannelIO(NATURE_CHANNEL_CRITTER_1 << 4 | NATURE_CHANNEL_CRITTER_3,
                                                  CHANNEL_IO_PORT_1, 1);
             }
```

Comment:
I wonder if it's worth some kind of comment to describe what's going on with these `Audio_SetNatureAmbienceChannelIO` function. Or maybe there's some kind of local macro that could be used? It's not immediately obvious from reading this code that setting port 1 to a val of 0 turns off that critter channel, while setting port 1 to a val of 1 turns it on.

What the value of a specific channel does is entirely sequence and channel dependent, so I don't know if there's a general purpose macro that's worth it or not.

Reply:
Is there hope that more audio documentation could help figure out a solution, including having enough documentation that it wouldn't be too hard for someone to figure it out themselves?

Reply:
I was also expecting this to be an improvement made on audios end. But I dont know enough about audio to say that its appropriate

Reply:
No worries, yeah I'm not sure exactly how to tackle these ambiences yet but it'll likely be after sequence 1 is decompiled. So we can leave it to the audio side of things for now then I suppose